EIPS/eip-3664.md

```diff
@@ -0,0 +1,417 @@
+---
+
+## Abstract
+In this proposal, we created the next gen component-NFT standard, which makes up for the problem that [ERC-721](./eip-721.md) and [ERC-1155](./eip-1155.md) can only support unchangeable attributes of NFTs. component-NFT can make the attributes in a NFT have five characteristics: transferable, upgradeable, changeable, combinable and evolvable. By using this standard NFTs in any fields, users can create any valuable dapps based on component-NFT standard.
+
+## Motivation
+In the early days of NFTs, ERC-721 was the main NFT protocol. It focused on differentiating NFTs from FTs. But since the emergence of the Layer 2 and Metaverse, outdated NFT protocols would no longer suffice for NFT standards in the new era. We hope to objectize NFTs, and allow for their upgrades, assembly and disaggregation on the blockchain. We will also let them change with the passage of time so that these fundamental functionalities will endow NFTs with a world of possibilities, creating unparalleled value.
+
+The NFT standard proposed by EIP-3664 enable unlimited disaggregation and assembly. Players can forge modularized NFTs with this proposal, and aggregate modularized NFTs into a new NFT, and let new NFTs combine with each other. On the other hand, NFTs can be disaggregated into component parts and reassembled. These powerful functionalities can sustain many different use cases in the future, such as team configuration in competitive gaming, amendable terms and clauses in financial contracts and honor awarding mechanisms in blockchain games. With the present proposal, you can realize any arbitrary functionality with your NFT. At the same time, the present proposal is an extension protocol of ERC-1155 and ERC-721, implying compatibility with ERC-721 and ERC-1155, and enabling the aggregation of NFTs on old protocols into newly modularized ones. 
+
+## Specification
+
+ERC-3664 extends ERC-1155, divides NFT attributes into five categories:
+
+1. General changeable attributes: usually refers to basic attributes such as ATK and DEF. Its functions include increase or decrease in attribute value.
+2. Transferable attributes: refers to the NFT's transferable attributes can be transferred to other NFTs in the event of destruction or other circumstances.
+3. Upgradeable attributes: refers to the NFT that can be upgraded according to the formula, the player pays the fee, and it is reflected in the increase of the level, usually accompanied by the increase of other attribute values.
+4. Combinable, NFTs can be combined with other NFTs, to make all NFT world combinable and separatable, each NFT is a component of other NFTs, freely combine to make greater value.
+5. Evolvable attribute: refers to the attribute that NFT may evolve automatically over time (the height of the chain block), simulate the time attributes in the real world, so that NFT has the characteristics of time-varying.
+
+```solidity
+function remove(uint256 _nftId, uint256 _attrId) external;
```

Comment:
I think this EIP could be structured better. I am not sure what these functions do and how the interfaces interact with them.

Many EIPs have chosen to describe the interface as a natspec documented Solidity interface. While I think we should not tie specification to Solidity, not better alternative have shown up yet.

Reply:
Thanks for your suggestions, we will convert the format the natspec standard interface asap.